js/http.js

```diff
@@ -240,6 +240,11 @@ var ERMrest = (function (module) {
                                     //throw new Error("httpUnauthorizedFn Event Handler not registered");
                                     deferred.reject(response);
                                 }
+                            } else {
+                                 // Push the current call to _authroizationDefers by calling _onHttpAuthFlowFn
+                                module._onHttpAuthFlowFn().then(function() {
```

Comment:
This is the code that we added for handling 401 authentication handling in ermrestjs. Now all unauthorised calls are handled by http layer in ermrestjs and chaise registers a callback which is called in case of http 401 errors. This callback does the login flow and returns a promise. On Successful completion of the flow and resolution of promise, this code will fire all  the calls which were delayed or failed due to the 401 error.